src/adm/tiles.rs

```diff
@@ -248,17 +249,22 @@ pub async fn get_tiles(
         metrics.incr_with_tags("filter.adm.empty_response", Some(tags));
     }
 
-    let filtered: Vec<Tile> = {
-        let filter = state.filter.read().await;
-        response
-            .tiles
-            .into_iter()
-            .filter_map(|tile| {
-                filter.filter_and_process(tile, location, &device_info, tags, metrics)
-            })
-            .take(settings.adm_max_tiles as usize)
-            .collect()
-    };
+    let mut filtered: Vec<Tile> = Vec::new();
```

Comment:
Is this change necessary? Not that it's terribly important with `adm_max_tiles` being small but previously it only acquired the read lock once.

Reply:
IIRC, the problem is that `filter_and_process()` now returns a `Result<Option<...>>`. I can't just call `?` on it otherwise I'll lose the error. That said, I can break `filter` out of the loop fairly easily.

Reply:
My fault, I missed this in the diff. You could collect into `Vec<Result<..>>` but that doesn't work well with the `take`ing, so this change makes sense.